ArduCopter/mode_flip.cpp

```diff
@@ -55,25 +58,49 @@ bool ModeFlip::init(bool ignore_checks)
         return false;
     }
 
-    // capture original flight mode so that we can return to it after completion
+    // capture original flight mode and throttle so that we can return to it after completion
     orig_control_mode = copter.flightmode->mode_number();
+    _flip_start_throttle = get_pilot_desired_throttle();;
+
 
     // initialise state
     _state = FlipState::Start;
     start_time_ms = millis();
 
     roll_dir = pitch_dir = 0;
 
-    // choose direction based on pilot's roll and pitch sticks
+    // choose direction based on pilot's roll and pitch input
     if (channel_pitch->get_control_in() > 300) {
         pitch_dir = FLIP_PITCH_BACK;
     } else if (channel_pitch->get_control_in() < -300) {
         pitch_dir = FLIP_PITCH_FORWARD;
-    } else if (channel_roll->get_control_in() >= 0) {
+    } else if (channel_roll->get_control_in() > 300) {
         roll_dir = FLIP_ROLL_RIGHT;
-    } else {
+    } else if (channel_roll->get_control_in() < -300) {
         roll_dir = FLIP_ROLL_LEFT;
     }
+    
+    // if controls are neutral, then choose direction based on parameter settings
+    // this will default to a left flip if the flip_roll parameter is out of bounds (0-3).
+    else {
```

Comment:
```suggestion
    } else {
        // if controls are neutral, then choose direction based on parameter settings
        // this will default to a left flip if the flip_roll parameter is out of bounds (0-3).
```